sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -2338,6 +2337,8 @@ pi_result piDeviceGetInfo(pi_device Device, pi_device_info ParamName,
   }
   case PI_DEVICE_INFO_NAME:
     return ReturnValue(Device->ZeDeviceProperties->name);
+  case PI_DEVICE_INFO_HOMOGENEOUS_ARCH:
+    return ReturnValue(PI_TRUE);
```

Comment:
add comment that we only opportunistically return true, no real guarantees are given in the spec currently.

Reply:
Level Zero spec says following in [zeModuleCreate](https://one-api.gitlab-pages.devtools.intel.com/level_zero/core/api.html#zemodulecreate) description:

> The application must only use the module for the device, or its sub-devices, which was provided during creation.

If I understand it correctly, we can consider this as "real guarantees" that any Level Zero device "has homogeneous architecture" i.e. we can re-use a module created for the root device on sub- or sub-sub- devices. @bashbaug, am I right?

Reply:
I quoted this part in the comments https://github.com/intel/llvm/pull/5240/commits/e6ca4f905caeb1cac075827234b27cdd09f18a5c.

Reply:
> Note, I believe the link above should be [zeModuleCreate](https://spec.oneapi.io/level-zero/latest/core/api.html#zemodulecreate).

Oops... Thanks for catching this. I've fixed the link.

> This sounds correct but with one additional caveat: the module is associated with both a context and a device, so if the root device and the sub-device are in the same context then the module created for the root device can also be used by the sub-device, but if the root device and the sub-device are in different contexts then the module will still need to be created twice.

DPC++ runtime library takes the context into account. Relevant comment - https://github.com/intel/llvm/pull/5240#discussion_r806335702.